basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java

```diff
@@ -1452,6 +1456,15 @@ private void doContentPutInternal(RunData data, Context context, Properties reqP
 		String id = reqProps.getProperty(LTIService.LTI_ID);
 		String toolId = reqProps.getProperty(LTIService.LTI_TOOL_ID);
 
+		// Encrypt the lti_content secret(SAK-46566)
+		String newSecret = reqProps.getProperty(LTIService.LTI_SECRET);
+		if (LTIService.SECRET_HIDDEN.equals(newSecret)) {
+			reqProps.remove(LTIService.LTI_SECRET);
+			newSecret = null;
```

Comment:
setting this to null is seems redundant

Reply:
I agree, Should we change this line of code too then?

`	// Insert or edit
	public void doToolPut(RunData data, Context context) {
		String peid = ((JetspeedRunData) data).getJs_peid();
		SessionState state = ((JetspeedRunData) data).getPortletSessionState(peid);

		if (!ltiService.isMaintain(getSiteId(state))) {
			addAlert(state, rb.getString("error.maintain.delete"));
			switchPanel(state, "Error");
			return;
		}
		Properties reqProps = data.getParameters().getProperties();

		String newSecret = reqProps.getProperty(LTIService.LTI_SECRET);
		if (LTIService.SECRET_HIDDEN.equals(newSecret)) {
			reqProps.remove(LTIService.LTI_SECRET);
			newSecret = null;
		}`